shim.c

```diff
@@ -1262,8 +1262,9 @@ EFI_STATUS init_grub(EFI_HANDLE image_handle)
 	// default loader if we get a TFTP error or HTTP error.
 	if (!use_fb && (efi_status == EFI_INVALID_PARAMETER ||
 	                efi_status == EFI_NOT_FOUND ||
-	                // TODO: Add EFI_HTTP_ERROR when gnu-efi is updated to a version that includes it
-	                // efi_status == EFI_HTTP_ERROR ||
+					// TODO: Change to EFI_HTTP_ERROR when gnu-efi is updated to a version that includes it
+					// I already get error 35 on my test systems, even with this build of shim.
+	                efi_status == EFIERR(35) ||
```

Comment:
I think it would be nicer to handle this would be with a compatibility macro at the top of this source file (or even in shim.h):

```
// FIXME: Drop this when gnu-efi is updated to include de6f9259e8476495c78babbc25250a59de7f3942.
#ifndef EFI_HTTP_ERROR
#define EFI_HTTP_ERROR EFIERR(35)
#endif
```

Then you can make this conditional more normal and the compatibility macro can be dropped when gnu-efi is updated. It's already on the gnu-efi [shim-15.9 branch](https://github.com/rhboot/gnu-efi/tree/shim-15.9), but the submodule here hasn't been updated.

Reply:
Thank you, I've made this change.